src/sentry/rules/history/preview.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from datetime import timedelta
+
+from django.utils import timezone
+
+from sentry.models import Project
+from sentry.rules import rules
+from sentry.rules.history.base import TimeSeriesValue
+from sentry.utils.types import Any, Dict, Sequence
+
+PREVIEW_TIME_RANGE = timedelta(weeks=2)
+
+
+def preview(
+    project: Project,
+    conditions: Sequence[Dict[str, Any]],
+    filters: Sequence[Dict[str, Any]],
+    condition_match: str,
+    filter_match: str,
+    frequency_minutes: int,
+) -> Sequence[TimeSeriesValue]:
+    end = timezone.now()
+    start = end - PREVIEW_TIME_RANGE
+    hours = get_hourly_bucket(PREVIEW_TIME_RANGE)
+    hourly_buckets = [0] * hours
+
+    # must have at least one condition to filter activity
+    if len(conditions) == 0:
+        return None
+    # all the currently supported conditions are mutually exclusive
+    elif len(conditions) > 1 and condition_match == "all":
+        return [TimeSeriesValue(start + timedelta(hours=i), 0) for i in range(hours)]
+
+    activity_set = set()
+    for condition in conditions:
+        condition_cls = rules.get(condition["id"])
+        if condition_cls is None:
+            return None
+        # instantiates a EventCondition subclass and retrieves activities related to it
+        condition_inst = condition_cls(project, data=condition)
+        try:
+            activity_set.update(condition_inst.get_activity(start, end))
```

Comment:
What do we de-dupe on here?

Reply:
I think reappearance and regression both have the same activity type (Regression)? I could be wrong. But we would want them to be duplicates if they happen on the same event